### PR TITLE
Prompt to save query

### PR DESCRIPTION
**Pull request by jamesstout**
What does this implement/fix? Explain your changes.
---------------------------------------------------
When closing a tab, you are prompted to save the SQL in the query window.

Does this close any currently open issues?
------------------------------------------
#455 

Any relevant logs, error output, etc?
-------------------------------------

Any other comments?
-------------------
TODO: prompt when closing entire app.
Maybe add a ["Don't warn me again" checkbox?](https://github.com/Sequel-Ace/Sequel-Ace/issues/455#issuecomment-723237083)  - Done.

Under non-ARC, there is an over-released view when you click "save" from the prompt.

Where has this been tested?
---------------------------
**Devices/Simulators:** iMac19,1

**macOS Version:**: 10.15.7 (19H2)

**Sequel-Ace Version:** main latest

### CODE REVIEW

**Comment by Jason-Morcos** [comment]
How hard would it be if instead of offering to save the content, we just cache it somewhere (maybe automatically save it as a SQL file in our container, or something) and load it back the next time the user opens the tab? Maybe that has some major weaknesses, I'm thinking along the lines of having Ace let you "pickup where you left off"? Also, not sure how to handle users with the same connection open in multiple tabs... this current solution is probably simpler

File: Source/Other/Data/SPConstants.m [hunk 1]
```diff
@@ -30,6 +30,31 @@
...
+	dispatch_once(&onceToken, ^{
+		tableViewType = [@{
+			@"source"		: @(SPTableViewStructure),
+			@"content"		: @(SPTableViewContent),
+			@"customQuery"	: @(SPTableViewCustomQuery),
+			@"status"		: @(SPTableViewStatus),
+			@"relations"	: @(SPTableViewRelations),
+			@"triggers"		: @(SPTableViewTriggers),
+			@"SPTableViewInvalid": @(NSNotFound)
+		} copy]; // TODO: won't need this copy under ARC
```
Comment (Kaspik) [inline Source/Other/Data/SPConstants.m:50]: Switch this to Swift (we have StringExtension already) and you are ARC done :)

**Review by Kaspik — COMMENTED** [review]
Re: Jason's comment - do we save this to recent queries?

**Comment by jamesstout** [comment]
> Re: Jason's comment - do we save this to recent queries?

Yes, if it has been executed

**Comment by Kaspik** [comment]
What if we would just save the query to latest queries without executing it and we wouldn't need any of this then? :)

**Comment by Kaspik** [comment]
I know it would mean we throw a lot of your work here... but that seems easiest, simplest, user friendly (as they are used to recent queries) and also... the least amount of code?

**Comment by jamesstout** [comment]
> I know it would mean we throw a lot of your work here... but that seems easiest, simplest, user friendly (as they are used to recent queries) and also... the least amount of code?

Let me check it out...

**Comment by jamesstout** [comment]
> > I know it would mean we throw a lot of your work here... but that seems easiest, simplest, user friendly (as they are used to recent queries) and also... the least amount of code?
> 
> Let me check it out...

Seems to work very well.

**Comment by jamesstout** [comment]
I'll close this and go with the history method

**Pull request closed without merging**